### PR TITLE
Integrate dev into main

### PR DESCRIPTION
**Pull request by weissercn**

### CODE REVIEW

**Comment by weissercn** [comment]
Currently failing due to the following error. Investigating
File "/nas/ucb/constantinweisser/Influence-benchmark/influence_benchmark/RL/expert_iteration.py", line 148, in launch
    selected_trajectories = get_top_n_trajectories(trajectory_folder, self.num_chosen_trajectories, self.mode)

File: influence_benchmark/experiments/hyperparameter_sweep.py [hunk 1]
```diff
@@ -52,7 +52,7 @@ def train_loop(config=None):
             accelerate_config_path=accelerate_config_path,
             sft_script_path=sft_script_path,
             model_name=model_name,
-            num_gen_trajectories_per_state=config.num_gen_trajectories_per_state,
+            num_gen_trajectories_per_state=6,
```
Comment (carolius) [inline influence_benchmark/experiments/hyperparameter_sweep.py:55]: kind of depends which parameters we want to vary each run but we might not want to push version with 6 hardcoded

**Comment by carolius** [comment]
We might want to check if sft trainer still correctly trains on the old lora rather than generating a new one after your changes, it can behave strangely and fail silently in my experience

**Comment by weissercn** [comment]
Done
- added environment queue
- added comments 
- fixed sbatch issues (needed to specify 1 node)
- removing code redundancies
- added slurm preparation
- saving settings for run 
- changed some copy to deepcopy

File: influence_benchmark/experiments/KTO_experiment copy.py
Comment (micahcarroll) [inline influence_benchmark/experiments/KTO_experiment copy.py:1]: Do we want to merge a `KTO_experiment copy.py` file? Probably not

File: influence_benchmark/experiments/expert_iteration_experiment.py [hunk 2]
```diff
@@ -21,15 +21,24 @@
...
+    iterations = 6
     ignore_first_n_assistant_messages = 1  # Number of assistant messages to not train on
     run_name = None
-    devices = [0, 1, 2, 3]  # , 4, 5, 6, 7]
+    devices = [
+        0,
+        1,
+        2,
+        3,
+    ]  # GPUs used for generating trajectories. The GPUs used for training are specified in the accelerate_config.yaml file.
```
Comment (micahcarroll) [inline influence_benchmark/experiments/expert_iteration_experiment.py:40]:
```suggestion
    # GPUs used for generating trajectories. The GPUs used for training are specified in the accelerate_config.yaml file.
    devices = [0, 1, 2, 3]  
```

File: influence_benchmark/experiments/hyperparameter_sweep.py (hunk 1 above)
Comment (micahcarroll) [reply influence_benchmark/experiments/hyperparameter_sweep.py:55]: I guess we're doing this?

File: influence_benchmark/experiments/single_expert_iteration_experiment.py [hunk 3]
```diff
@@ -8,16 +8,19 @@
...
     num_gen_trajectories_per_state = 32
     num_chosen_trajectories = 4  # on a single GPU across all trajactories
     iterations = 5
     ignore_first_n_assistant_messages = 1  # Number of assistant messages to not train on
     run_name = None
-    devices = [1]
-    mode = "single"  # single environment
+    devices = [
+        2
+    ]  # GPUs used for generating trajectories. The GPUs used for training are specified in the accelerate_config.yaml file.
```
Comment (micahcarroll) [inline influence_benchmark/experiments/single_expert_iteration_experiment.py:22]:
```suggestion
    # GPUs used for generating trajectories. The GPUs used for training are specified in the accelerate_config.yaml file.
    devices = [2]  
```

File: influence_benchmark/stats/preferences_per_iteration.py [hunk 4]
```diff
@@ -1,86 +1,95 @@
...
+                n_trajectories=len(x), reward_avg_all_trajectories=x["expected_preference"].mean()
+            ).nlargest(num_chosen_trajectories, "expected_preference")
+        )
+        .reset_index(drop=True)
+    )
+    top_n = top_n.rename(
+        columns={
+            "expected_preference": "reward_avg_selected_trajectories",  # average after selecting trajectories # nlargest_[trajectory_id](mean_[turn](reward))
+        }
+    )
```
Comment (micahcarroll) [inline influence_benchmark/stats/preferences_per_iteration.py:50]: Not carefully reviewed, just going to assume it's correct

**Review by micahcarroll — APPROVED** [review]
LGTM! Let's wait to merge until the remaining SLURM stuff bugs are figured out and merged

**Pull request merged**